docs/custom-code/prisma-custom-attributes.md

```diff
@@ -0,0 +1,327 @@
+
+1. **Make sure usernames are unique**: When a user signs up for your application, you want to ensure that each user has a unique username.
+Using Prisma's custom attributes you can enforce this constraint at the field level.
+
+```
+model User {
+  username String @unique
+}
+```
+
+:::important
+This is a simplified Prisma schema to showcase the `@unique` custom attribute.
+:::
+
+2. **Auto generating IDs**: In many applications, it's important to have a unique ID for each record, but it can be error-prone to generate these manually.
+Prisma provides you with custom attributes to ensure each record has a unique, automatically-incremented ID.
+
+```
+model Customer {
+  id Int @id @default(autoincrement())
+}
+```
+
+You can see **two** field-level attributes used for the `id` field.
+
+The first is the `@id` custom attribute to indicate that the `id` field will act as the ID on the entity.
+
+The second is the `@default()` custom attribute function.
+`@default()` is used to define a default value for a field.
+`@default()` has a parenthesis associated with it indicating that you can pass parameters to it.
+In this case, the parameter is `autoincrement()`, a Prisma function that can auto-generate an ID.
+
+3. **Customize the database column that a field maps to**: You can customize the name of the table that can entity maps to in the database with Prisma's `@@map` custom attribute.
+
+This is useful when you want to map an entity (for example `Customer`) to a table with a different name that does not match model naming conventions (for example `customers`).
+
+```
+model Customer {
+ id Int @id @default(autoincrement())
+ createdAt DateTime @default(now())
+ firstName String
+ 
+ @@map("website_customers")
+}
+```
+
+These are just a few examples of what Prisma's Custom Attributes allow you to do.
+
+In the following sections, we're going to dive deeper into the nuances of the syntax, which custom attributes are available for you, and what they do.
+
+## How to add Custom Attributes to your Entities and Fields
+
+Amplication provides you with a text field in the UI known as the **Custom Attributes** text field.
+
+This text field is available in 2 places: 
+
+1. In the _Field Settings_ screen, for adding custom attributes to your individual Fields.
+2. In the Entity's _General Settings_ page, for adding custom attributes to your Entities as a whole.
+
+The custom attributes that you use for _Fields_ and the ones you use for _Entities_ are different syntactically and functionally.
+
+Let's dive into the specifics of each with in-depth explanations and examples.
+
+## How to add Field-level Custom Attributes
+
+Field-specific Custom Attributes are prefixed with a single `@`.
+
+You have access to custom attributes, like `@id`, and also custom attribute functions like `@default()`.
+
+### Field-level Custom Attributes
+
+Here are the attributes that you have access to on the field level:
```

Comment:
we should mention that the user can use anything that is supported by Prisma and add alink to Prisma doc rather then listing the available options here since it may change and we don't want to keep track and updating on our side